### PR TITLE
[QC-619] Force the facility name to be less than 32 characters

### PR DESCRIPTION
**Pull request by Barthelemy**

### CODE REVIEW

File: Framework/include/QualityControl/runnerUtils.h [hunk 1]
```diff
@@ -81,6 +81,20 @@ inline int computeRunNumber(const framework::ServiceRegistry& services, const bo
   return run;
 }
 
+/**
+ * If the string is longer than the requested length, truncate it.
+ * @param toBeTruncated
+ * @return toBeTruncated truncated to `length` characters
+ */
+inline std::string truncate(std::string toBeTruncated, size_t length)
```
Comment (knopers8) [inline Framework/include/QualityControl/runnerUtils.h:89]: Doesn't `string::substr()` do exactly that? It should survive length() > length

**Review by knopers8 — APPROVED** [review]

File: Framework/include/QualityControl/runnerUtils.h (hunk 1 above)
Comment (Barthelemy) [reply Framework/include/QualityControl/runnerUtils.h:89]: I thought not but let me give it a try !

Comment (knopers8) [reply Framework/include/QualityControl/runnerUtils.h:89]: Returns a substring [pos, pos+count). If the requested substring extends past the end of the string, or if count == npos, the returned substring is [pos, size()). 
from https://en.cppreference.com/w/cpp/string/basic_string/substr

Comment (Barthelemy) [reply Framework/include/QualityControl/runnerUtils.h:89]: Dammit, you were right !
Fixed;

**Pull request merged**